cmd/kubeadm/app/phases/kubeconfig/kubeconfig.go

```diff
@@ -96,26 +80,47 @@ func createKubeConfigFileForClient(masterEndpoint, client, outDir string, caCert
 	)
 
 	// Write it now to a file
-	filepath := path.Join(outDir, fmt.Sprintf("%s.conf", client))
-	return WriteKubeconfigToDisk(filepath, config)
+	filename := filepath.Join(outDir, fmt.Sprintf("%s.conf", client))
+	return WriteKubeconfigToDisk(filename, config)
 }
 
-func WriteKubeconfigToDisk(filepath string, kubeconfig *clientcmdapi.Config) error {
-	// Make sure the dir exists or can be created
-	if err := os.MkdirAll(path.Dir(filepath), KubernetesDirPermissions); err != nil {
-		return fmt.Errorf("failed to create directory %q [%v]", path.Dir(filepath), err)
+func WriteKubeconfigToDisk(filename string, kubeconfig *clientcmdapi.Config) error {
+	// Convert the KubeConfig object to a byte array
+	content, err := clientcmd.Write(*kubeconfig)
+	if err != nil {
+		return err
+	}
+
+	// Create the directory if it does not exist
+	dir := filepath.Dir(filename)
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		if err = os.MkdirAll(dir, KubernetesDirPermissions); err != nil {
+			return err
+		}
 	}
 
-	// If err == nil, the file exists. Oops, we don't allow the file to exist already, fail.
-	if _, err := os.Stat(filepath); err == nil {
-		return fmt.Errorf("kubeconfig file %s already exists, but must not exist.", filepath)
+	// Check if the file exist, and if it does check if the current content matches the expected content
+	if _, err := os.Stat(filename); err == nil {
```

Comment:
Can you invert this to indent the error flow here, e.g. : 
```
if _, err := os.Stat(filename); err != nil {
    WriteFile()
}
```
https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow